include/nbl/builtin/hlsl/binops.hlsl

```diff
@@ -0,0 +1,25 @@
+#ifndef _NBL_BUILTIN_HLSL_BINOPS_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BINOPS_INCLUDED_
+
+namespace nbl
+{
+namespace hlsl
+{
+
+#define COMPOUND_ASSIGN(NAME,OP) template<typename T> struct assign_ ## NAME ## _t { \
+    void operator()(NBL_REF_ARG(T) lhs, NBL_CONST_REF_ARG(T) rhs) \
+    { \
+        lhs = lhs OP rhs; \
+    } \
+}
+COMPOUND_ASSIGN(add,+);
+COMPOUND_ASSIGN(subtract,-);
+COMPOUND_ASSIGN(mul,*);
+COMPOUND_ASSIGN(div,/);
+
+#undef COMPOUND_ASSIGN
```

Comment:
add them here https://github.com/Devsh-Graphics-Programming/Nabla/blob/66332959a4b5df61dc1e327dfa646fa91ea00de1/include/nbl/builtin/hlsl/functional.hlsl#L67

and make sure they have names `plus_assign`, `multiplies_assign` to be inline with the rest.

Also because its just 4 struct defs and not specialization, IMHO you shouldn't use macros because that just makes the code iffy to read. Or if you are to use a macro make sure it has the same form as the `ALIAS_STD` so you have to put th e `};` afterwards to be able to add extra fields (i.e. identity)

Reply:
To be able to define an identity for, say, complex numbers, I'd need the templated expressions `T(0)` or `T(1)` to make sense when replacing `T` by `complex_t<T>`. Is that doable?